app/models/user.place.server.ts

```diff
@@ -30,36 +30,27 @@ export const mutation = {
     user,
     place,
     placeImage,
+    placeQRCode,
   }: {
     user: Pick<User, "id">;
     place: Pick<Place, "name" | "description">;
     placeImage: Pick<PlaceImage, "url">;
+    placeQRCode: Pick<PlaceQRCode, "url">;
   }) {
-    // TODO: refactor into a single transaction
-    const newPlace = await prisma.place.create({
+    return prisma.place.create({
       data: {
         user: { connect: { id: user.id } },
         slug: createPlaceSlug(place),
         name: place.name.trim(),
         description: place.description.trim(),
+        qrCode: {
+          create: { url: placeQRCode.url, user: { connect: { id: user.id } } },
+        },
+        images: {
+          create: [{ url: placeImage.url, user: { connect: { id: user.id } } }],
```

Comment:
With the help of Codeium, I found the issue that caused us stuck last night during #23. 😅

We were missing `user: { connect: { id: user.id } }`, because each `image` requires connecting to the `user` entity. 😓

Reply:
haha That's why I tend to make the fields as optional in the beginning